Documentation/cluster-crd.md

```diff
@@ -17,6 +17,7 @@ Settings can be specified at the global level to apply to the cluster as a whole
 
 - `name`: The name that will be used internally for the Ceph cluster. Most commonly the name is the same as the namespace since multiple clusters are not supported in the same namespace.
 - `namespace`: The Kubernetes namespace that will be created for the Rook cluster. The services, pods, and other resources created by the operator will be added to this namespace. The common scenario is to create a single Rook cluster. If multiple clusters are created, they must not have conflicting devices or host paths.
+- `finalizers`: Specify the `cluster.rook.io` finalizer to allow the Rook operator to properly clean up resources when the cluster resource is being deleted. Otherwise, volume attachments may not be cleaned up properly before the cluster pods and resources are removed.
```

Comment:
is the a strong reason to expose this to the user?  i.e. are there any scenarios where a user wouldn't want the rook finalizer to run?  this feels like a detail that should be abstracted away from the user, such that the operator would manage it without user intervention.  Anytime a cluster object is created, the operator could insert the finalizer into the object.

Reply:
yes, that would be better to automatically add it. We don't have a pattern of adding properties to the crd, but this seems like the right time.